### DISCUSSION TITLE
The  "__CUDA_ARCH_LIST__" macro dosen't work on Jetson Xavier NX with nvcc 11.4

### DISCUSSION THREAD

**Discussion by Lurvelly** [NVIDIA/cuCollections#262] [General]
Hi All,

 I am trying to use cuCollections library on my Jetson Xavier NX with customer key,  but get the error that said "build/_deps/cuco-src/include/cuco/static_map.cuh(157): error: static assertion failed with "A key type larger than 8B is supported for only sm_70 and up."

As all known, the Jetson Xavier NX is Volta and support sm_72, so I double checked my CMakelist.txt, I am sure I have added these " --generate-code arch=compute_72,code=sm_72" in the CMAKE_CUDA_FLAGS.

Then I found that the "__CUDA_ARCH_LIST__" macro dosen't work, that's why there is no define "CUCO_HAS_INDEPENDENT_THREADS". 

https://github.com/NVIDIA/cuCollections/blob/2dbcfc7a4798d4a5f3a12b74bf4a3e1cd236e834/include/cuco/detail/__config#L22
https://github.com/NVIDIA/cuCollections/blob/2dbcfc7a4798d4a5f3a12b74bf4a3e1cd236e834/include/cuco/detail/__config#L32-L34

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_May__4_00:02:26_PDT_2022
Cuda compilation tools, release 11.4, V11.4.239
Build cuda_11.4.r11.4/compiler.31294910_0

**Comment by PointKernel** [comment]
Thanks for reporting the issue.

Related to #245 

IIRC, `__CUDA_ARCH_LIST__` is supported by nvcc 11.5 or plus. @Lurvelly Is it possible to install 11.5 on your Jetson?

cc @sleeepyjack

**Reply by Lurvelly** [reply]
Hi, @PointKernel 

Thanks for your reply.

I will try to check if my Jetson can upgrade the nvcc than 11.5. If not, I would like to use the `#define CUCO_CUDA_MINIMUM_ARCH 720` to replace current macro at https://github.com/NVIDIA/cuCollections/blob/2dbcfc7a4798d4a5f3a12b74bf4a3e1cd236e834/include/cuco/detail/__config#L22

It seems to be the simplest way to solve my issue, isn't it?  If I did,  will there be another issue in the `cuCollections` ?

**Reply by PointKernel** [reply]
> I would like to use the #define CUCO_CUDA_MINIMUM_ARCH 720 to replace current macro

Yeah, that should work just fine. (Somehow I remember a valid CUDA arch has 2 digits only, so it would be either `720` or `72`, yours to discover!)

> will there be another issue in the cuCollections?

Hopefully not, do let us know if you have any further issues. :smile:

**Reply by Lurvelly** [reply]
> Yeah, that should work just fine. (Somehow I remember a valid CUDA arch has 2 digits only, so it would be either 720 or 72, yours to discover!)

 I have double checked the CUDA arch on my computer, it should be 720. 

> Hopefully not, do let us know if you have any further issues. 

Ok, I will test the cuCollections on my Jetson, hope it works fine. And if I meet another issue about cuCollections, I will let you know asap.

Thanks again for your reply.

Best Regards
---------
Louis Li
